### ISSUE TITLE
[Expressions] Boolean functions output 1/0 instead of true/false

### ISSUE THREAD

**Issue by DelazJ** [qgis/QGIS#39433]
An expression calling the boolean version of overlay_intersects, overlay_within... function displays in the preview true or false as output.
OTOH, Boolean expressions using operators, geometric predicate as intersects, within... usually return 1 or 0.
Might be good harmonizing.

**Comment by nyalldawson** [comment]
I'd fix the others - they have legacy behaviour only because they predated bool support in the expression engine

**Comment by Pedro-Murteira** [comment]
@DelazJ Hello, is this still valid on recent releases?

**Comment by DelazJ** [comment]
Still valid (I updated the title to reflect Nyall's comment). Boolean functions show 1/0 in preview (and i guess in output) and not true/false. Things are even more complicated now because their help text states they return TRUE/FALSE.

**Comment by phidrho** [comment]
@DelazJ 
is this still present?
We've had bug fix which seems that has never been implemented.

**Comment by DelazJ** [comment]
Yes, still there. From 3.43 (look at the preview and what the help says)

![Image](https://github.com/user-attachments/assets/135bafa5-aa66-40c3-97d2-e528e9b2d868)
